### PR TITLE
Avoid logger.fatal on agent forwarding errors

### PR DESCRIPTION
**Pull request by carkmorwin**
In attempt to keep the agent running and avoid hard crashes, we return
client errors from forwarding when encountered. This wil cause the
goroutine for that file to exit, but keep the agent running serving
other behaving files where client config is correct.

On server errors, unless we drop the logs, we exit and assume the
process will restart and retry the failures. A more robust solution
should be introduced be will not be at this time.

### CODE REVIEW

File: forwarding.go [hunk 1]
```diff
@@ -41,23 +43,29 @@ func Forward(bufChan chan *bytes.Buffer, httpClient *retryablehttp.Client, endpo
 
 		resp, err := httpClient.Do(req)
 		if err != nil {
-			// retries have already happened at this point, so give up
+			// we have failed all of our retries at this point
 			if !discard {
 				logger.Fatal(err)
```
Comment (binarylogic) [inline forwarding.go]: Why are we calling `Fatal` here? Shouldn't we indefinitely retry until successful instead of exiting?

Comment (carkmorwin) [reply forwarding.go]: @binarylogic we now wait forever

Comment (binarylogic) [reply forwarding.go]: 👍

**Comment by jralph** [comment]
Great, thanks! Any ETA on this being released and the agent/docker image being updated?

Perhaps a solution that monitors the endpoint and pauses all collection of logs on any downtime and waits until healthy to continue?

**Comment by jralph** [comment]
Any news on this being released at all? We'd like to get our timber agent running again but can't until this is live!

**Comment by carkmorwin** [comment]
@jralph Sorry for the delay on this. `v0.9.0` just went out and includes a fix for this issue. That release includes our binaries as well as a docker container. Enjoy!

**Comment by jralph** [comment]
@carkmorwin Thanks for the release! We've got it running now but we're experiencing really really high cpu usage in 0.9.0 compared to 0.8.*. 0.9.0 seems to be using considerably more cpu (over 100x more).

**Comment by carkmorwin** [comment]
@jralph I'll take a look. There is a place where a syscall was introduced that I'm assuming is the culprit but will confirm.

**Comment by carkmorwin** [comment]
@jralph Filed an issue that you can track. #161

**Pull request merged**